### ISSUE TITLE
Smartline state collecter constantly says blocked

### ISSUE THREAD

**Issue by Alatarius** [joe7575/techpack#123]
The smartline state collector keeps saying blocked even though I have 1 battery, 1 smartline controller and 1 smartline state collector. I know nothing about coding so I need a step-by-step eli5 explanation.

Everything is programmed(?) like it says in the wiki, and functions are not working.

This is the code copied from the output tab:

1 (0s): IF inp(0101,is off) THEN tower(0097,green) after 0s
2 (0s): IF inp(0101,is on) THEN tower(0097,red) after 0s
3 (0s): IF fuel(0015,is empty) THEN tower(0097,amber) after 0s
4 (0s): IF fuel(0015,is loaded) THEN tower(0097,green) after 0s
5 (s): IF ... THEN ... after s
6 (s): IF ... THEN ... after s
7 (s): IF ... THEN ... after s
8 (s): IF ... THEN ... after s

![Untitled-1](https://github.com/joe7575/techpack/assets/101660302/e6cb0440-50ce-46f9-a257-0d8b984c0ca9)
![Untitled-2](https://github.com/joe7575/techpack/assets/101660302/66657ba6-6bcc-47df-9344-87aa795113db)

**Comment by joe7575** [comment]
Thats easy :)
The smartline state collector outputs the state with the highest priority from the "connected" blocks.
At least one of the connected blocks is in the state "blocked".

**Comment by Alatarius** [comment]
Nothing is blocked. I just set it up and don't even have it running yet. I have built this thing before and it works fine but the items here to monitor it don't work. If a machine gets the defect and needs to be fixed, it won't change the smartlight to red like programmed and the pushers each side is running at a different time.

![Untitled](https://github.com/joe7575/techpack/assets/101660302/d1f816c2-0a1e-4ca6-b155-3bd9c869c9a2)

None of the graphics are lined up on any side. Also switches have no effect even when block numbers are correctly inputed.

**Comment by Alatarius** [comment]
I just tested this on a new world with this mod as the only one active with the same results

**Comment by joe7575** [comment]
I tried again, the collector does exactly what it's supposed to do. So I can't understand your problem

**Comment by Alatarius** [comment]
I don't get it either, but every time I add a second smartline controller, all the collectors say blocked. If I remove all but 1 smartline controller then the collector says running.

**Comment by joe7575** [comment]
>  but every time I add a second smartline controller or more then all the collectors say blocked.
This sentence irritates me.
Normally **one** collector collects the states from several machines and provides the summarized result to **one**  smartline controller.
What does your structure look like?

**Comment by Alatarius** [comment]
collector is monitoring these nodes:
0015 0016 0006 0023 0008 0024 0025 0026 0027 0010 0089 0090 0091 0092 0093 0354 0347 0356

controller #1:
1 (0s): IF inp(0101,is off) THEN tower(0097,green) after 0s
2 (0s): IF inp(0101,is on) THEN tower(0097,red) after 0s
3 (0s): IF fuel(0015,is empty) THEN tower(0097,amber) after 0s
4 (0s): IF fuel(0015,is loaded) THEN tower(0097,green) after 0s
5 (0s): IF inp(0090,is off) THEN tower(0087,green) after 0s
6 (0s): IF inp(0090,is on) THEN tower(0087,red) after 0s
7 (0s): IF fuel(0024,is empty) THEN tower(0024,red) after 0s
8 (0s): IF fuel(3,0024 loaded) THEN tower(000024,green) after 0s

controller #2
1 (0s): IF inp(0101,is off) THEN tower(0097,green) after 0s
2 (0s): IF inp(0101,is on) THEN tower(0097,red) after 0s
3 (0s): IF fuel(0354,is empty) THEN tower(0024,red) after 0s
4 (0s): IF fuel(0354,is loaded) THEN tower(0024,green) after 0s

**Comment by Alatarius** [comment]
the collectors or 0101 and 0090

**Comment by joe7575** [comment]
0090 is the collector number and at the same time part of the monitor list of the collector?
This will not work
